catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -115,81 +123,135 @@
+          new UpdateStorageRequestImpl(contentUpdates, reprocessProperties);
+      try {
+        catalogFramework.update(updateStorageRequest);
+      } catch (IngestException | SourceUnavailableException e) {
+        LOGGER.debug("Error storing thumbnail/overview/original", e);
+      }
+    }
 
-  private void process(List<ContentItem> contentItems) {
-    List<ContentItem> newContentItems = new ArrayList<>();
-    contentItems.forEach(contentItem -> process(contentItem, newContentItems));
-    contentItems.addAll(newContentItems);
+    if (!metacardUpdates.isEmpty()) {
+      UpdateRequest updateRequest =
+          new UpdateRequestImpl(
+              metacardUpdates
+                  .stream()
+                  .map(
+                      mcard ->
+                          new AbstractMap.SimpleEntry<Serializable, Metacard>(mcard.getId(), mcard))
+                  .collect(Collectors.toList()),
+              Core.ID,
+              reprocessProperties);
+      try {
+        catalogFramework.update(updateRequest);
+      } catch (IngestException | SourceUnavailableException e) {
+        LOGGER.debug("Error updating metacard thumbnail", e);
+      }
+    }
   }
 
-  private void process(ContentItem contentItem, List<ContentItem> contentItems) {
-    Metacard metacard = contentItem.getMetacard();
+  private boolean shouldGenerateContentItems(
+      Metacard metacard, Map<String, Serializable> properties) {
+    Attribute type = metacard.getAttribute(Media.TYPE);
+    return type != null
+        && IMAGE_NITF.equals(type.getValue())
+        && !(boolean) properties.getOrDefault(NITF_PROCESSING_KEY, false);
+  }
 
-    if (!isNitfMimeType(contentItem.getMimeTypeRawData())) {
-      LOGGER.debug(
-          "skipping content item: filename={} mimeType={}",
-          contentItem.getFilename(),
-          contentItem.getMimeTypeRawData());
+  private void generateImages(
+      Metacard metacard, List<Metacard> metacardUpdates, List<ContentItem> contentUpdates) {
+    ResourceResponse response;
+    try {
+      response = catalogFramework.getLocalResource(new ResourceRequestById(metacard.getId()));
+    } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+      LOGGER.debug("Error retrieving resource for thumbnail/overview/original creation", e);
       return;
     }
 
-    try {
-      if (contentItem.getSize() / MEGABYTE > maxNitfSizeMB) {
+    byte[] originalThumbnail = metacard.getThumbnail();
+
+    int contentCount = contentUpdates.size();
+    process(metacard, response.getResource().getInputStream(), contentUpdates);
+
+    if (contentCount == contentUpdates.size() && metacard.getThumbnail() != originalThumbnail) {
+      metacardUpdates.add(metacard);
+    }
+  }
+
+  private void process(Metacard metacard, InputStream input, List<ContentItem> contentItems) {
+    try (InputStream source = input) {
+      if (getResourceSize(metacard) / MEGABYTE > maxNitfSizeMB) {
```

Comment:
In theory, `getResourceSize` could fail to parse the resource size into a long; should we protect against that?

Reply:
We are also calling `getResourceSize(metacard) / MEGABYTE` twice

Reply:
Changed the method to `getResourceSizeInMB` and added a separate catch block for the NumberFormatException.